core/services/wifi/WifiManager.py

```diff
@@ -128,11 +128,13 @@ async def get_saved_wifi_network(self) -> List[SavedWifiNetwork]:
         except Exception as error:
             raise FetchError("Failed to fetch saved networks list.") from error
 
-    async def add_network(self, credentials: WifiCredentials) -> int:
+    async def add_network(self, credentials: WifiCredentials, hidden: bool = False) -> int:
         """Set network ssid and password
 
         Arguments:
             credentials {[WifiCredentials]} -- object containing ssid and password of the network
+            hidden {bool} -- Should be set as true when adding hidden networks (that have SSID as
+            blank or null). False by default.
```

Comment:
Can't we just do it if WifiCredentials.ssid is empty ?

Reply:
You're not mixing hidden with unprotected?

Reply:
WifiCredentials has ssid information, if ssid is empty so the network is in hidden mode, no ?

Reply:
This is the add_network method. To add the network, we need it's SSID. The SSID is just not available on the scan, but we still need it, so in this case, it's provided by the user on the frontend (when the frontend detects a hidden network, it shows an input box for the SSID, togheter with the one for the password, and sends, together with the WifiCredentials, a hidden flag.

Reply:
> This is the add_network method. To add the network, we need it's SSID.

So the comment in the code is wrong: `that have SSID as blank or null`
It should be: `that does not broadcast the SSID`

Reply:
Better. Changing.